cli/command/stack/opts.go

```diff
@@ -24,6 +24,12 @@ func addRegistryAuthFlag(opt *bool, flags *pflag.FlagSet) {
 	flags.BoolVar(opt, "with-registry-auth", false, "Send registry authentication details to Swarm agents")
 }
 
+func addNoImageResolutionFlag(opt *bool, flags *pflag.FlagSet) {
+	var flagNoResolveImage = "no-resolve-image"
+	flags.BoolVar(opt, flagNoResolveImage, false, "Do not query the registry to resolve image digest and supported platforms")
+	flags.SetAnnotation(flagNoResolveImage, "version", []string{"1.30"})
+}
```

Comment:
A minor nit: typically we only use these functions for flags that are shared between commands. I think it's fine to keep this as a separate function, but maybe move it to the single file where it's used?